### DISCUSSION TITLE
Glitches when encoding certain files

### DISCUSSION THREAD

**Discussion by LazyNCoder** [staxrip/staxrip#794] [Q&A]
Hello,

When I try to encode some files from a certain TV network, I get glitches(green screen for >1 sec). I don't know what might cause this but it started to happen while ago and never got fixed. I don't know if this is a x265 problem or Stax's

I have an old build of stax and x265 which doesn't have this problem. When I update x265, it start to appear but strangely, when I use my old x265 on a new version of staxrip, it would also appear.

Example file:
`Room.104.S04E12.1080p.WEB.H264-CAKES`
Also happens when some of other files from the same TV network which I don't wanna name here.

How to reproduce:
Just get the latest version of Staxrip, encode using vapoursynth and everything default settings. 
The first frames of video will be totally green frames.

Please note, this is not a problem with the source file. I can totally encode this using Handbrake or my older version of Staxrip, and everything is fine. 

I appreciate if anyone could help.

**Comment by LazyNCoder** [comment]
I noticed, changing decoder from avisynth/vapoursynth to ffmpeg(intel) resolves the issues. But I don't know what's the consequences of this choice. Is it something big? Does it affect performance or quality? Any opinions?

**Comment by Patman86** [comment]
Do you still use filters when encoding?
Which settings do you use in x265?
How did you update StaxRip? Copy paste does not work.

**Comment by LazyNCoder** [comment]
I faced the problem over a year ago. I waited and tried each new version of Staxrip hoping it would eventually get fixed. 
So, each time I would download a new copy to test it. To reproduce it, you just need the file I mentioned(happened on other similar files too). 
Open a fresh copy of Staxrip, change to Vapoursynth and drag drop the file on Staxrip. no need to change filters or settings or anything. Click on preview and press on the right arrow key on your keyboard to go to the next frame. There's the ugly green frames.
There's nothing wrong with the file. If you use  LWLibavSource, there would be no issue. I also tried with other software like hadbrake, and the result is fine too.

**Comment by Patman86** [comment]
I have now tried to reproduce this according to your statements and explanations, but unfortunately without success. I used the file `Room.104.S04E12.1080p.WEB.H264-CAKES` for the test and set everything up as you described. Now when I start the preview, I see an image without glitches.

You do all of this over the network and not from a local hard drive, did I read that correctly?

**Comment by LazyNCoder** [comment]
First, thank you very much for pursuing this. 
I have to ask, did you click on "forward 1 frame" on the preview window? the first frame is fine.

**Reply by Patman86** [reply]
Yes, I tested it "forward 1 frame" with the keyboard and the mouse, both possibilities bring a fine first frame.

**Comment by LazyNCoder** [comment]
I got a little (1min) video clip for you to check:
https://gfycat.com/femaleinnocentamericanwigeon

**Reply by Patman86** [reply]
Here is my record

[https://a.tmp.ninja/poSCfJMr.mp4](https://a.tmp.ninja/poSCfJMr.mp4)

here with same file as you

[https://a.tmp.ninja/fPeQZcq.mp4](https://a.tmp.ninja/fPeQZcq.mp4)

and here same way as you

[https://a.tmp.ninja/eqXFpOR.mp4](https://a.tmp.ninja/eqXFpOR.mp4)

**Reply by LazyNCoder** [reply]
Very strange indeed. Do you have any clue what could be the cause of this on my end? I tried this on a fresh install of windows as well. Same results. It's fine with LWLibavSource tho.

**Reply by Patman86** [reply]
Please install this package and test whether it helps

[VCRedist AIO](https://github.com/abbodi1406/vcredist/releases/latest)

**Reply by LazyNCoder** [reply]
> 
> 
> Please install this package and test whether it helps
> 
> [VCRedist AIO](https://github.com/abbodi1406/vcredist/releases/latest)

Unfortunately, this didn't help. I really thought it would, but it didn't. I even redownloaded the ffms dlls from the github but still no luck. There should be a way tho, because you didn't have any problems. Something's not right....
Thank you very much anyways ♥

**Reply by Patman86** [reply]
graphic card?

pls install [DirectX](https://www.microsoft.com/en-us/download/details.aspx?id=8109) and [LAVFilters](https://github.com/Nevcairiel/LAVFilters/releases/latest)

if that doesn't help either, I don't know what to do next 🤔

**Comment by LazyNCoder** [comment]
I just checked. For me, this happens on AVS as well(on both FFMS and FMVideoSource), no need to switch to Vapoursynth.

**Comment by LazyNCoder** [comment]
I got a clue. downgrading ffms to a previous version(2.23.1) did resolve the problem. But I'm not too fine with older versions(since it had bugs and stuff).
Do you think by any chance, your stax loads any old version of FFMS from like syswow64 or any other places?

I'm gonna try to install the DX and LAVFilters as well. I'll let you know

**Reply by Patman86** [reply]
my stax uses all plugins that are included in the package, no separate plugins or other places

latest version of ffms2 by STvG (mod) [ffms2](https://ppp.woelkli.com/s/bx8zFHStJT9aCqs) (pass: tTnkia1ToVkw)

latest version of ffms2 by Myrsloik (dev) [ffms2](https://www.dropbox.com/s/1u8r4ia5phatjgx/ffms3000-test1.7z?dl=1)

**Reply by LazyNCoder** [reply]
The STvG mod was identical with the files that ship with Stax. Tried the other one and still it wouldn't help.
Thank you very much for everything. 
By the way, which one do you prefer? Using FFMS2 2.23.1 or LWLibavSource?

I would investigate this further and if I got any leads, I'll post here.

**Reply by Patman86** [reply]
Use the one with which you have fewer problems, i.e. LWLibavSource. With LWLibavSource you are also more flexible.
It's definitely weird ... It's stupid if the error is not really reproducible. Please feel free to keep me up to date.
